torch/csrc/jit/interpreter.cpp

```diff
@@ -393,6 +794,15 @@ struct CodeImpl {
     list.size++;
   }
 
+  void aliasRegistersTo(ArrayRef<Value*> new_allocations, ArrayRef<Value*> existing_allocations) {
```

Comment:
Invariant: this needs to be called before any of `existing_allocations` are used, right?

Reply:
It needs to be called before any `new_allocations` are used. The assertion for this being valid is in the loop:
```
JIT_ASSERT(unique_to_reg.count(e) > 0 && unique_to_reg.count(n) == 0);
```